File: HospiotalServiceHub/IServices/IPatientService.cs
using HospitalDataModel.ViewModel;

namespace HospiotalServiceHub.IServices
{
    public interface IPatientService
    {
        Task<IEnumerable<PatientViewModel>> GetAllPatients(int skip, int take);

        Task<PatientViewModel> GetPatient(long id); 

         Task<bool> CreatePatient(PatientViewModel patient);

        Task<bool> DeletePatient(long id);

        Task<bool> UpdatePatient(PatientViewModel patient);

        //add for test
        bool IsEven(int number);
    }
}

File: hospitalapp/src/interfaces/IPatient.service.ts
import IPatientModel from "../models/IPatient.model";


interface IPatientService {
    GetAllPatients(skip:number ,take: number): Promise<IPatientModel[] | null>;

    GetPatientById(id: number): Promise<IPatientModel>;

    CreatePatient(model: IPatientModel): Promise<void>;

    UpdatePatient(model: IPatientModel): Promise<void>;

    DeletePatient(id: number): Promise<void>;
}

export default IPatientService;